core/transcoder.go

```diff
@@ -145,6 +145,16 @@ func NewNvidiaTranscoder(gpu string) TranscoderSession {
 	}
 }
 
+func NewNvidiaTranscoderWithDetector(detector ffmpeg.DetectorProfile, gpu string) (TranscoderSession, error) {
+	// Hardcode detection to device 0 for now
+	// Transcoding can still run on a separate GPU as we copy frames to CPU before detection
+	session, err := ffmpeg.NewTranscoderWithDetector(detector, "0")
```

Comment:
Just a note that can be addressed outside of this PR:

I think this API is a bit confusing. While I know that the device ID arg is only relevant for initializing the DNN filtergraph, when I first read this function my first thought is that the device ID would be used for both the transcoder and the DNN filtergraph.